src/cli/build.rs

```diff
@@ -55,20 +55,46 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     let conda_build_noop = NoopCondaBuildReporter::new();
     // Construct a temporary directory to build the package in. This path is also
     // automatically removed after the build finishes.
+    let pixi_dir = &project.pixi_dir();
+    tokio::fs::create_dir_all(pixi_dir)
+        .await
+        .into_diagnostic()
+        .with_context(|| {
+            format!(
+                "failed to create the .pixi directory at '{}'",
+                pixi_dir.display()
+            )
+        })?;
     let work_dir = tempfile::Builder::new()
         .prefix("pixi-build-")
         .tempdir_in(project.pixi_dir())
         .into_diagnostic()
         .context("failed to create temporary working directory in the .pixi directory")?;
 
+    // Build platform virtual packages
+    let build_platform_virtual_packages: Vec<GenericVirtualPackage> = project
```

Comment:
Why wouldn't it matter? It also needs to know the vpkgs for the build setup right? Or is that overwritten later?

Reply:
I tested with only host and it tells me it can't install the `build environment`:

<details>
<summary>pixi build output with only host virtual packages</summary>
```
 pixi build

 ╭─ Running build for recipe: navigator-0.0.1-hbf21a9e_0
 │
 │ ╭─ Fetching source code
 │ │ No sources to fetch
 │ │
 │ ╰─────────────────── (took 0 seconds)
 │
 │ ╭─ Running cache build
 │ │
 │ ╰─────────────────── (took 0 seconds)
 │
 │ ╭─ Resolving environments
 │ │ 
 │ │ Resolving build environment:
 │ │   Platform: linux-64 [__unix=0=0, __linux=4.18=0, __glibc=2.28=0, __archspec=1=x86_
 │ │ 64]
 │ │   Channels: 
 │ │    - file:///home/rarts/dev/ros_workspace/src/navigator/.pixi/pixi-build-Qoao3p/
 │ │    - conda-forge
 │ │    - robostack-staging
 │ │   Specs:
 │ │ 
 │ │ ╭──────────────────────────┬─────────┬──────────────┬─────────────┬────────────╮
 │ │ │ Package                  ┆ Version ┆ Build        ┆ Channel     ┆ Size       │
 │ │ ╞══════════════════════════╪═════════╪══════════════╪═════════════╪════════════╡
 │ │ │ _libgcc_mutex            ┆ 0.1     ┆ conda_forge  ┆ conda-forge ┆ 2.50 KiB   │
 │ │ │ _openmp_mutex            ┆ 4.5     ┆ 2_gnu        ┆ conda-forge ┆ 23.07 KiB  │
 │ │ │ binutils_impl_linux-64   ┆ 2.43    ┆ h4bf12b8_1   ┆ conda-forge ┆ 5.94 MiB   │
 │ │ │ binutils_linux-64        ┆ 2.43    ┆ h4852527_1   ┆ conda-forge ┆ 34.09 KiB  │
 │ │ │ gcc_impl_linux-64        ┆ 14.1.0  ┆ h3c94d91_1   ┆ conda-forge ┆ 69.69 MiB  │
 │ │ │ gcc_linux-64             ┆ 14.1.0  ┆ h3f71edc_4   ┆ conda-forge ┆ 31.18 KiB  │
 │ │ │ gxx_impl_linux-64        ┆ 14.1.0  ┆ h8d00ecb_1   ┆ conda-forge ┆ 14.31 MiB  │
 │ │ │ gxx_linux-64             ┆ 14.1.0  ┆ hc55ae77_4   ┆ conda-forge ┆ 29.59 KiB  │
 │ │ │ kernel-headers_linux-64  ┆ 3.10.0  ┆ he073ed8_17  ┆ conda-forge ┆ 922.94 KiB │
 │ │ │ ld_impl_linux-64         ┆ 2.43    ┆ h712a8e2_1   ┆ conda-forge ┆ 653.92 KiB │
 │ │ │ libgcc                   ┆ 14.2.0  ┆ h77fa898_1   ┆ conda-forge ┆ 828.85 KiB │
 │ │ │ libgcc-devel_linux-64    ┆ 14.1.0  ┆ h5d3d1c9_101 ┆ conda-forge ┆ 2.63 MiB   │
 │ │ │ libgomp                  ┆ 14.2.0  ┆ h77fa898_1   ┆ conda-forge ┆ 450.19 KiB │
 │ │ │ libsanitizer             ┆ 14.1.0  ┆ hcba0ae0_1   ┆ conda-forge ┆ 4.32 MiB   │
 │ │ │ libstdcxx                ┆ 14.2.0  ┆ hc0a3c3a_1   ┆ conda-forge ┆ 3.71 MiB   │
 │ │ │ libstdcxx-devel_linux-64 ┆ 14.1.0  ┆ h5d3d1c9_101 ┆ conda-forge ┆ 13.39 MiB  │
 │ │ │ sysroot_linux-64         ┆ 2.17    ┆ h4a8ded7_17  ┆ conda-forge ┆ 14.44 MiB  │
 │ │ │ tzdata                   ┆ 2024b   ┆ hc8b5060_0   ┆ conda-forge ┆ 119.49 KiB │
 │ │ ╰──────────────────────────┴─────────┴──────────────┴─────────────┴────────────╯
 │ │ 
 │ │ Resolving host environment:
 │ │   Platform: linux-64 []
 │ │   Channels: 
 │ │    - file:///home/rarts/dev/ros_workspace/src/navigator/.pixi/pixi-build-Qoao3p/
 │ │    - conda-forge
 │ │    - robostack-staging
 │ │   Specs:
 │ │    - ros-humble-ament-cmake *
 │ │    - ros-humble-ament-index-cpp *
 │ │    - ros-humble-rclcpp *
 │ │    - ros-humble-rclcpp-action *
 │ │    - ros-humble-std-msgs *
 │ │    - ros-humble-geometry-msgs *
 │ │    - ros-humble-turtlesim *
 │ │    - ros2-distro-mutex ==0.5 humble
 │ │    - cmake *
 │ │    - ninja *
 │ │    - libstdcxx >=14
 │ │    - libgcc >=14
 │ │
 │ ╰─────────────────── (took 1 second)
 │
 ╰─────────────────── (took 1 second)
  × during the building of the project the following error occurred
  ├─▶ Failed to resolve dependencies: Cannot solve the request because of: ros-humble-
  │   turtlesim * cannot be installed because there are no viable options:
  │   ├─ ros-humble-turtlesim 1.4.2 | 1.4.2 would require
  │   │  └─ __glibc >=2.17,<3.0.a0, for which no candidates were found.
  │   └─ ros-humble-turtlesim 1.4.2 | 1.4.2 would require
  │      └─ qt-main >=5.15.6,<5.16.0a0, which cannot be installed because there are no
  │   viable options:
  │         └─ qt-main 5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 |
  │   5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 |
  │   5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 |
  │   5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 would require
  │            └─ __glibc >=2.17,<3.0.a0, for which no candidates were found.
  │   
  ╰─▶ Cannot solve the request because of: ros-humble-turtlesim * cannot be installed
      because there are no viable options:
      ├─ ros-humble-turtlesim 1.4.2 | 1.4.2 would require
      │  └─ __glibc >=2.17,<3.0.a0, for which no candidates were found.
      └─ ros-humble-turtlesim 1.4.2 | 1.4.2 would require
         └─ qt-main >=5.15.6,<5.16.0a0, which cannot be installed because there are no
      viable options:
            └─ qt-main 5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 |
      5.15.6 | 5.15.6 | 5.15.6 | 5.15.6 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 |
      5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 |
      5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 | 5.15.8 would require
               └─ __glibc >=2.17,<3.0.a0, for which no candidates were found.
```

</details>

Reply:
Thats a bug in the backend then, because it should use the detect virtual packages. I think it makes sense to use the detected virtual packages for the build environment rather than the minimum requirements from the manifest. But I could be wrong ..

Reply:
Yes, oke lets just leave this as is.